engine/consensus/matching/core_test.go

```diff
@@ -149,14 +151,13 @@ func (ms *MatchingSuite) TestOnReceiptPendingReceipt() {
 	// the mempool, and return false as we are testing the case where it was already in the mempool
 	ms.ReceiptsPL.On("AddReceipt", receipt, ms.UnfinalizedBlock.Header).Return(false, nil).Once()
 
-	// OnReceipt should return immediately after realizing the receipt is already in the mempool
+	// onReceipt should return immediately after realizing the receipt is already in the mempool
 	// but without throwing any errors
-	err := ms.matching.OnReceipt(receipt.ExecutorID, receipt)
+	_, err := ms.matching.onCurrentReceipt(receipt)
 	ms.Require().NoError(err, "should ignore already pending receipt")
 
 	ms.ReceiptsPL.AssertExpectations(ms.T())
 	ms.ReceiptsDB.AssertNumberOfCalls(ms.T(), "Add", 0)
-	ms.ResultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
```

Comment:
Very good observation!

The reason I remove it is because I don't like to mock the mempool modules. We should be using a real mempool module in the tests, [as I started doing here](https://github.com/onflow/flow-go/pull/387/files#diff-fb1b89101e5c9e4bba14378cb7978c2210ae8c80b3e3a4933d00552ccdfdc5b5R72). 

I think we should mock the "external" dependency (i.e. receiptsValidator), rather than internal modules. The results mempool is only being updated by the matching engine, using a mock would make it more often to break when making changes. 

The tests could test the behavior of the engine, and maybe assert on the data in the mempool, rather than checking how many times a function of the mempool being called, the error message would also be confusing.

If the tests couple too much on the internal implementation, it's more subject to change, which is the case here:  Even if a receipt is a duplication, I always add to the result mempool; but the behavior actually didn't change, as after calling onReceipt, the receipt should be stored in the mempool.